app/components/toast_message_component.html.erb

```diff
@@ -0,0 +1,19 @@
+<div class="sm:rounded-md shadow-sm p-4 mb-8 bg-<%=- @color %>-50">
+  <div class="flex">
+    <div class="flex-shrink-0">
+      <%= inline_svg_tag icon_path(icon: @icon), class: "h-5 w-5 text-#{@color}-400" %>
```

Comment:
Just a heads up, the tailwind purger won't correctly purge classes that are spliced together like this. It's annoying, but you'd have to clearly declare text-red-400, text-green-400, etc. for the purger to pick up on it, rather than concatenating strings together

Reply:
I noticed there was at least one other spot where the tailwind class was concatenated like this, so just thought I'd mention it!

Reply:
Oh thanks for pointing this out. I'm not super familiar with tailwind yet. Do you have any suggestions?

Reply:
Great catch, @Tonksthebear.

@lsparlin, instead of passing in only the `color`, we should pass in the entire class name. For example, `text-red-400` or `text-blue-400`. Then we can make sure that the Tailwind config is checking `app/components/` for class names to purge.

Reply:
I'm not sure if it's proper, but I normally just define a list of constant class colors in the ruby file, such as:

`STYLE = {
  blue: 'text-blue-400',
  green: 'text-green-400'
}`

Then I just pass in the style I want to the component like you already have set up. Just instead of setting @color the parameter, I'd do
`@color = STYLES[color]`

Then swap `text-#{@color}-400` with `@color`

Hopefully that makes sense! I wrote this up pretty quickly... there might be a better way, just wanted to get some ideas out there in case they're useful.

Reply:
There are several such classes in the component. You think there should be a parameter for each of them?

Reply:
Yeah, this could get ugly quickly, @lsparlin. I wonder what other apps are doing in this space.

Reply:
I definitely understand how it becomes unwieldy. I just couldn't find another way because Tailwind requires such a verbose syntax. To accommodate multiple parameters, I just make the hash larger and define getters. So some quick sudo code for what I do

```
STYLES = {
  blue: {
    header: 'text-blue-200',
    message: 'text-blue-400',
    footer: 'text-blue-100'
  }
...
def header_classes
  STYLES[@color][:header]
end

def message_classes
  STYLES[@color][:message]
end

def footer_classes
  STYLES[@color][:message]
end
```

Then I would just call the corresponding getters from the html.

Again, I completely get how verbose this solution may sound but this is what worked best for my mental model when working with Tailwind

Reply:
So as long as the class (full literal string) shows up anywhere in `html` or `rb` files tailwind will be happy?

Thank you @Tonksthebear

Reply:
Not to overload this thread with more information, but another avenue I've pursued is creating a helper that pulls from a config file for the corresponding classes.

For instance, I'd have a config/styles.yml file that contains something like

```
toast:
  blue:
    header: 'text-blue-200 ...'
    message: 'text-blue-400 ...'
    footer: 'text-blue-100 ...'
```

Then in the html I'd just call `tailwind(toast: { @color => :header })`

I've enjoyed this pattern due to no boilerplate in the .rb files, but it does feel dangerously close to just simulating css classes. I suppose another avenue would be using ViewComponent's sidecar assets in some way (if possible). 

In either case, it's a problem I've been personally wrestling with so I'd love to hear other solutions

Reply:
> So as long as the class (full literal string) shows up anywhere in `html` or `rb` files tailwind will be happy?
> 
> Thank you @Tonksthebear

Yup! Just have to update the tailwind.config.js to point it to the correct files

Reply:
My vote is for a big hash and helper methods for now. But I do love a YAML-powered configuration for the different types of colors! Maybe if we need them in the future that will be a good route to take.